js/server/leader.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief active failover leadership change
+// /
+// / @file
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2014 ArangoDB GmbH, Cologne, Germany
+// / Copyright 2011-2014 triAGENS GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// /
+// / @author Allen Plum
+// / @author Copyright 2021, ArangoDB GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+(function(){
+  "use strict";
+require("@arangodb/foxx/manager").healAll();
```

Comment:
it should be run in all, so that the routes become installed in all v8-contexts; else they will 404 and not list in aardvark on first try, which was another facette of the original bug.

Reply:
Got it. IMO it would be good to add locks around the self-heal procedure because when executed concurrently, it can lead to undefined behavior. But this has been the case even before this PR so it should be attacked separately.

Reply:
No, this only needs to run in one. The reload routing is going to tell all other contexts to purge their routing maps and reload the Foxx services. @jsteemann is right.

Reply:
@jsteemann in the case of `threadNumber === 0` probably the c++ wraper needs to fetch the special context from the dealer? Is this available with `--console` ?

Reply:
The V8 context used for `--console` will be #1, not #0. So I guess it is safe to restrict the healAll call to `threadNumber === 0`:
```
if (require("internal").threadNumber === 0) { 
  require("@arangodb/foxx/manager").healAll(); 
}
```